src/OnlineRetailPortal.Core/Models/Product.cs

```diff
@@ -53,12 +54,14 @@ public async Task<Product> SaveAsync(IProductStore productStore, ProductConfigur
             return addProductResponse.ToModel();
 
         }
-
         public async Task<Product> UpdateAsync(IProductStore productStore)
         {
-            var productEntity = this.ToStoreEntity();
-            var updateProductResponse = await productStore.UpdateProductAsync(productEntity);
+            var getProductResponse = await productStore.GetProductAsync(this.Id);
```

Comment:
getProductResponse is a verb where as a variable which is storing ProductResponse can not have any action of its own. this can be renamed to ProductResponse

Reply:
done